### ISSUE TITLE
SC2002 calls 'cat $var' useless 

### ISSUE THREAD

**Issue by brchrisman** [koalaman/shellcheck#2333]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2002
- My shellcheck version (`shellcheck --version` or "online"): 0.7.2
- [ X ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [ X ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ X ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ X ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
cat $foo | wc -l

```

#### Here's what shellcheck currently says:

    ^-- SC2002: Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.

#### Here's what I wanted or expected to see:
Calling cat with multiple files is valid and doesn't get flagged by SC2002.
Calling cat with a variable containing two whitespace delimited filenames could be intended.

In this case, one suggestion would change output and the other would result in an error:
```
#!/bin/bash
files="a b"
cat $files | wc -l
```
Or if possible... disabling SC2086 could disable this warning for cat-with-variable, because if you want to use word splitting after variable evaluation, then you don't want this check?

Awesome tool, btw!

**Comment by koalaman** [comment]
You're right, it should offer SC2086 or SC2002 but not both. This is now fixed. Thanks!

There is a common and unfortunate regression in `x=foo.txt; cat $x | ..` where SC2086 is smart enough to realize the variable won't split, but SC2002 is not, thus causing both suggestions to be suppressed. This will require more of a rewrite, but is not a huge concern.
